src/Raven.Studio/typescript/viewmodels/database/tasks/editReplicationHubTask.ts

```diff
@@ -0,0 +1,480 @@
+            {
+                content:
+                    "<ul class='margin-bottom margin-bottom-xs padding'>" +
+                        "<li><small>These prefixes define what docments are allowed to be <strong>sent to this Hub.</strong></small></li>" +
+                        "<li><small>You can <strong>further restrict this list</strong> when defining a Sink task that sends data to this Hub.</small></li>" +
+                    "</ul>"
+            });
+    }
+
+    importCertificate(fileInput: HTMLInputElement) {
+        this.spinners.importCertificate(true);
+        fileImporter.readAsText(fileInput,data => this.certificateImported(data));
+        this.spinners.importCertificate(false);
+    }
+
+    certificateImported(cert: string) {
+        try {
+            this.editedReplicationAccessItem().certificate(new replicationCertificateModel(cert));
+            this.editedReplicationAccessItem().usingExistingCertificate(true);
+        } catch ($e) {
+            messagePublisher.reportError("Unable to import certificate", $e);
+        }
+    }
+    
+    generateCertificate() {
+        app.showBootstrapDialog(new generateReplicationCertificateConfirm())
+            .done(validity => {
+                if (validity != null) {
+                    this.spinners.generateCertificate(true);
+                    const editedItemBefore = this.editedReplicationAccessItem();
+                    
+                    new generateCertificateForReplicationCommand(this.activeDatabase(), validity)
+                        .execute()
+                        .done(result => {
+                            const editedItemAfter = this.editedReplicationAccessItem();
+                            
+                            if (editedItemBefore === editedItemAfter) {
+                                this.editedReplicationAccessItem().certificate(new replicationCertificateModel(result.PublicKey, result.Certificate));
+                               
+                                // reset the 'saving certificate' status
+                                this.editedReplicationAccessItem().accessConfigurationWasExported(false);
+                                this.editedReplicationAccessItem().certificateWasDownloaded(false);
+                                this.editedReplicationAccessItem().usingExistingCertificate(false);
+                            }
+                        })
+                        .always(() => this.spinners.generateCertificate(false));
+                }
+            });
+    }
+    
+    exportHubConfiguration() {
+        if (!this.isValid(this.editedHubTask().validationGroupForExport)) {
+            return;
+        }
+        
+        this.exportConfiguration();
+    }
+
+    exportAccessConfiguration() {
+        if (!this.isValid(this.editedHubTask().validationGroupForExport)) {
+            return;
+        }
+        
+        if (this.editedHubTask().withFiltering()) {
```

Comment:
same here - we could move this logic to editedReplicationAccessItem and avoid repeating ourselves.